src/main/kotlin/com/openlattice/linking/graph/PostgresLinkingQueryService.kt

```diff
@@ -291,16 +291,20 @@ class PostgresLinkingQueryService(private val hds: HikariDataSource, private val
         }
     }
 
-    override fun getEntityKeyIdsOfLinkingIds(linkingIds: Set<UUID>): PostgresIterable<Pair<UUID, Set<UUID>>> {
+    override fun getEntityKeyIdsOfLinkingIds(
+            linkingIds: Set<UUID>,
+            normalEntitySetIds: Set<UUID>?
+    ): PostgresIterable<Pair<UUID, Set<UUID>>> {
         return PostgresIterable(
                 Supplier {
                     val connection = hds.connection
-                    val ps = connection.prepareStatement(ENTITY_KEY_IDS_OF_LINKING_IDS_SQL)
+                    val ps = connection.prepareStatement(buildEntityKeyIdsOfLinkingIdsSql(normalEntitySetIds != null))
                     val linkingIdsArray = PostgresArrays.createUuidArray(connection, linkingIds)
 
-                    val partitions = PostgresArrays.createIntArray( connection, getPartitionsInfo( linkingIds, partitionManager.getAllPartitions() ))
-                    ps.setArray(1, partitions)
```

Comment:
is there a reason why we don't need/can't use partition id here anymore?

Reply:
We used to compute the partition for linked data based on the linking id, but we recently switched to using the origin entity key id to compute the partition. In order to accurately compute the partition for the linking data, we'd need to look up the normal entity key id (origin id) for the linking ids, which is what this function is doing.

The one case where we *could* use the partition is in the case where all the normalEntitySetIds all have a singleton partition list where the partition is the same between all of them, but that seems like a rare/specific enough case that it's not worth special casing.

Reply:
👍 I think we probably want it since it's part of the primary key for `IDS` table

Reply:
We can use all the partitions of the normal entity set ids, since the actual partitions are a subset of that (and it's still better than going through all partitions)

Reply:
We had a similar discussion with Matthew about how to handle `PostgresEntityDataQueryService.getEntitySetIterable` for linking = true cases and it seemed the most efficient to use all entity set partitions.
https://jira.openlattice.com/browse/LATTICE-2319

Reply:
Yeah, it depends how this is being called i.e. many normal entity sets are passed here / how many partitions they belong to -- if they end up spanning pretty much all the partitions anyway the extra partitions lookup will just be extra overhead, but if not then it might speed up the query. I can add it in, but I imagine it won't be strictly an improvement on perf.

Reply:
(I added it, but we may want to keep an eye on typical usage here)